src/components/spreadsheet/table-wrapper.js

```diff
@@ -0,0 +1,686 @@
+            props.equipmentType !== null &&
+            !manualTabSwitch
+        ) {
+            const newIndex = getTabIndexFromEquipementType(props.equipmentType);
+            setTabIndex(newIndex); // select the right table type
+            // calculate row index to scroll to
+            const rows = getRows(newIndex);
+            let index = rows.findIndex((r) => r.id === props.equipmentId);
+            setScrollToIndex(index ? index : undefined);
+        } else if (manualTabSwitch) {
+            setScrollToIndex();
+        }
+    }, [
+        props.network,
+        props.equipmentId,
+        props.equipmentType,
+        props.equipmentChanged,
+        getRows,
+        manualTabSwitch,
+    ]);
+
+    useEffect(() => {
+        if (scrollToIndex) {
+            gridRef.current.api?.ensureIndexVisible(scrollToIndex, 'top');
+        }
+    }, [gridRef, scrollToIndex]);
+
+    useEffect(() => {
+        const lockedColumnsConfig = TABLES_DEFINITION_INDEXES.get(tabIndex)
+            .columns.filter((column) => lockedColumnsNames.has(column.id))
+            .map((column) => {
+                return { colId: column.field, pinned: 'left' };
+            });
+
+        if (isEditColumnVisible()) {
+            lockedColumnsConfig.unshift({ colId: 'edit', pinned: 'left' });
+        }
+
+        gridRef.current?.columnApi?.applyColumnState({
+            state: lockedColumnsConfig,
+            defaultState: { pinned: null },
+        });
+    }, [isEditColumnVisible, lockedColumnsNames, tabIndex]);
+
+    const handleColumnDrag = useCallback(
+        (event) => {
+            if (event.finished && event.column) {
+                const [reorderedItem] = reorderedTableDefinitionIndexes.splice(
+                    reorderedTableDefinitionIndexes.indexOf(
+                        event.column.colDef.id
+                    ),
+                    1
+                );
+                const destinationIndex = isEditColumnVisible()
+                    ? event.toIndex - 1
+                    : event.toIndex;
+
+                reorderedTableDefinitionIndexes.splice(
+                    destinationIndex,
+                    0,
+                    reorderedItem
+                );
+                setReorderedTableDefinitionIndexes(
+                    reorderedTableDefinitionIndexes
+                );
+
+                updateConfigParameter(
+                    REORDERED_COLUMNS_PARAMETER_PREFIX_IN_DATABASE +
+                        TABLES_NAMES[tabIndex],
+                    JSON.stringify(reorderedTableDefinitionIndexes)
+                ).catch((error) => {
+                    snackError({
+                        messageTxt: error.message,
+                        headerId: 'paramsChangingError',
+                    });
+                });
+
+                const [reorderedColDef] = columnData.splice(
+                    columnData.indexOf(
+                        columnData.find((obj) => {
```

Comment:
Array.prototype.findIndex() exists
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

Reply:
Implemented it